### ISSUE TITLE
1/6/2024 What's important to you (the user)

### ISSUE THREAD

**Issue by Neriderc** [Neriderc/GVExport#487]
It would be good to hear what features are important to anyone using GVExport, to help prioritise the order that things are added.

Feel free to mention existing issues that you'd like prioritised, or to suggest things that have not yet been mentioned.

Small things or big things are welcome. You can comment on little things that annoy you, options you wish you had, or big features on your wish list.

**Comment by schuco** [comment]
GVExport is able to display very complicated relationships between family branches. Sometimes, however, the diagrams become very confusing. I find it particularly annoying when Graphviz places person boxes between siblings in a family to save space. I think that it is hardly possible to "prevent" Graphviz, as a general program that is not specialized in genealogy, from doing this. However, I have seen that Graphviz can also display 3D graphics in VRML format. I have no idea and am asking quite naively whether it would be possible to move the annoying boxes that have been inserted into sibling groups for reasons of space to a lower level. Then, however, it should still be possible to display the 3D graphics without special 3D glasses or software.

**Comment by Neriderc** [comment]
> GVExport is able to display very complicated relationships between family branches. Sometimes, however, the diagrams become very confusing. I find it particularly annoying when Graphviz places person boxes between siblings in a family to save space. I think that it is hardly possible to "prevent" Graphviz, as a general program that is not specialized in genealogy, from doing this. 

I also don't much like how Graphviz sometimes groups people. In the past I have investigated trying to resolve this, and the options seem to be 1. to increase the weighting for some links ("edges", in DOT terms), which I've found to give me inconsistent results that I can't claim are any better, or option 2, use groupings to effectively force items to appear together. However, you can't just do groups but have to actually assign the row/column for the group. In doing this, we lose the advantage of Graphvix handling this for us. We simply do not know where particular groups should fall before we have traversed the tree. It just isn't practical to use this method as we lose a major reason for using Graphviz to start with.

 > However, I have seen that Graphviz can also display 3D graphics in VRML format. I have no idea and am asking quite naively whether it would be possible to move the annoying boxes that have been inserted into sibling groups for reasons of space to a lower level. Then, however, it should still be possible to display the 3D graphics without special 3D glasses or software.

I think the problem doesn't go away with this, because we aren't telling Graphviz to insert non-siblings in the middle of siblings, so we can't really change our input to tell it which boxes to put on another layer. 

I have thought about this a bit, and even if we did find a way to force group siblings, I don't think we would want that. Sometimes the siblings each have hundreds of descendants in the tree and should be spaced far apart to allow this to fit.

Most family tree software only lists descendants or ancestors of a specific person. We are trying to display all information in the family tree, even if they aren't a direct ancestor or descendant, and sometimes even if they are not related except distantly through marriage. Have you seen other software that manages to do this and do a good job of it? It would be good to see some inspiration.

**Comment by schuco** [comment]
@Neriderc   Have you seen other software that manages to do this and do a good job of it?

I have not and that is the reason why I like and use GVExport.
Thank you that you are again open for any ideas, even diletantic ones like mine, to further improve this great tool!
As you say, let us wait to see some inspiration of others

**Comment by elysch** [comment]
Hello all.

The lines are hard to follow when GVExport generates a big diagram. It would be great to use different kinds of lines in a random manner (kinds of dashes or dots or widths, I don't know). This way not all the lines look the same.
[https://graphviz.org/docs/attr-types/style/](Graphviz style)

I guess it wouldn't be easy to do and I'm not sure if it would look weird.

What do you think?

**Comment by Neriderc** [comment]
> The lines are hard to follow when GVExport generates a big diagram. It would be great to use different kinds of lines in a random manner

It seems it's not too difficult to do. I just did a little trial:
![screen shot of GVE diagram with joining lines using dots, dashes, tapered, or solid lines](https://github.com/user-attachments/assets/e23774ee-029a-41da-9e9b-d4eba913634b)

I know this isn't a big complicated diagram so might be hard to really judge if it helps, but let me know what you think.

**Comment by elysch** [comment]
It looks ok to me... I can test it if you tell me what to change.

**Comment by Neriderc** [comment]
I've just pushed my changes to the "random-lines" branch, so if you download that you can test it out. It's only a quick change, there is no option to disable the setting yet! If it looks like what you're after then I can add the option to the UI and all that ancillary stuff.

**Comment by elysch** [comment]
Maybe something is missing?. I tried it and all lines look the same as before.

**Comment by Neriderc** [comment]
Oh sorry, I hadn't committed the changes! I have done it now, so it should work if you try again.

**Comment by elysch** [comment]
I don't know what others think, but I liked it very much.

💯

**Comment by Neriderc** [comment]
I'll make a new issue to track this and add it as an option :slightly_smiling_face:

**Comment by elysch** [comment]
Another idea. To show a message with the number of individuals and families when you generate a download file and add it to the list of messages. Just like when you click on the "View" button. That way there is no need to view before download to know this information.

**Comment by Neriderc** [comment]
That should be easy enough to do, I'll make a new issue to track that as well.

**Comment by Neriderc** [comment]
> Would it be possible to display the expected dimensions of a PDF diagram in GVExport?

I have done some testing, I can't get it exact but it seems I can get it within a few MM.

The DPI is dots per inch. The SVG generated in the browser includes a height and width we could use, where the height and width are measured in "points" or "pt", which is basically equivalent to one of the dots in the DPI. i.e. divide the number of pt by the DPI to get the number of inches.

Graphviz seems to add on a margin that by my quick testing seems to be 10mm each side (total 20mm extra width) and perhaps 12 or 13 mm top and bottom though it came back slightly different in different tests (total around 24-26mm). It's probably doing some level of rounding that means it won't be an exact figure. Perhaps it's not 10mm but 0.4 inches or something like that. I haven't tried different DPI either.

Using this information I seem to be able to get close to the dimensions shown when inspecting a downloaded PDF generated on the server. 

How accurate do you need it for it to be useful?

**Comment by Neriderc** [comment]
> I love all the different options. Since a marriage is displayed, would it also be possible to display a divorce? At the moment I would like to display marriages but it is a bit confusing if someone is already divorced but this is not visible.

I don't see any reason why not. Perhaps with this divorce symbol?  ⚮

I can create a new issue to track this and will add at some point in the future.

**Comment by Neriderc** [comment]
> Another Idea, which would be beneficial for me: Option to additionally show the current family name (last married name without divorce) in brackets beside the birth name as surname. Since some of the family members may not know the birth name and are therefore irritated.

GVExport is not showing the "birth" name, but rather it's asking webtrees for the "primary" name. In most cases, the primary name is simply the name listed at the top of the list of names for a person. If you want to show a different name for someone, add it in webtrees on the individual's page, and make sure it's the top one. Click the "edit" button on the individual's page, and choose to "reorder names". You can then drag the names to put the one you want to display in GVExport at the top of the list.

**Comment by Neriderc** [comment]
> Another point that I noticed: Is it possible to add an option to also display stepchildren?

In what context are they not displayed? If I select someone with a step child, and choose to show 1 generation of descendants, this shows the step child even if no other parents are shown.

**Comment by elysch** [comment]
In my case the problem was that I was adding a fact to the child, but I never added the child to the family as adopted.

Look at this webtrees forum post: https://www.webtrees.net/index.php/forum/2-open-discussion/40071-how-to-find-the-stepchildren-from-a-parent#111615

I still think there should be an option to add the "Adoption" text to arrows in GVExport. Right?

**Comment by elysch** [comment]
Oh, I see. 

I think that color should not be used In the random arrow colors then, otherwise it's impossible to discern.

**Comment by hartenthaler** [comment]
This are two options 
- blood-related: this can be used to indicate or show all persons related by blood to at least one of the starting persons 
- color of arrow between a person and the family that person belongs to as a child. 

If you use the second option the color depends if the child belongs to the family (not the parents) by birth or by adoption. There are more relationship types (like foster or rada), but I have to test the behavior for them.

**Comment by hartenthaler** [comment]
In GEDCOM and webtrees a family has two parent persons (named HUSB and WIFE). So there is nothing special for the color of the arrows from the parents to the family node. 

A child can have several parent families. The relationship from a child to any of its parent families is indicated by birth or adoption or foster or rada or ...

The adoption link can be "adopted by both parents" or "adopted by HUSB" or "adopted by WIFE". HUSB and WIFE are mostly male and female, but this is not necessary. As far as I know "foster" is always related to both parents. "Rada" for the giving milk wife is by definition always related to the WIFE person. What do you expect how this should be visualized?

**Comment by hartenthaler** [comment]
The "adopted by HUSB" is a special case of a relation between two persons (not person to family or family to person). There are more such relationships like the godfather or godmother relationships. Or the relationship between an employer and an employee. They are indicated in GEDCOM by ASSO or _ASSO. This allows to build up social networks instead of ancestry trees. There are for example large trees of doctor fathers in mathematics available. Leonard Euler was the doctor father of x. X was the doctor father of y. And so on. Relationships spanning more than 1,000 years. 

It would be interesting to visualize such a ASSO tree using GVExport. There is no possibility in webtrees to visualize ASSO trees.

**Comment by elysch** [comment]
> The adoption link can be "adopted by both parents" or "adopted by HUSB" or "adopted by WIFE". HUSB and WIFE are mostly male and female, but this is not necessary. As far as I know "foster" is always related to both parents. "Rada" for the giving milk mother is by definition always related to the WIFE person. What do you expect how this should be visualized?

That's why I was adding a label to the arrow from father to family... could be also from mother to family or both. Now that you mention it, Maybe it's better to put the label only from family to child specifying if was adopted by HUSB or WIFE or BOTH. That way there can be different kinds of adoption for each child. 

The other kinds of relationships between children and families, I guess that could be handled the same way: with a label in the arrow from family to child.

**Comment by Neriderc** [comment]
Sorry I've been AWOL, thanks for the pull request! So the things covered in the conversation are solved with the pull request?

**Comment by elysch** [comment]
The pull request, if enabled, adds one label to the arrow from the family to the individual with the pedigree, when it's different than birth.

In case it's an adoption, it uses the 3 ADOP to be more specific

**Comment by elysch** [comment]
Changed the pull request to: https://github.com/Neriderc/GVExport/pull/570

**Comment by Neriderc** [comment]
I'm going to close this as it has got long again. Feel free to open new issues with new requests!
